app/src/lib/stores/alive-store.ts

```diff
@@ -0,0 +1,217 @@
+  }
+
+  private unsubscribeFromAllAccounts() {
+    const subscribedAccounts = this.subscriptions.map(s => s.account)
+    for (const account of subscribedAccounts) {
+      this.unsubscribeFromAccount(account)
+    }
+  }
+
+  private subscribeToAccounts = (accounts: ReadonlyArray<Account>) => {
+    if (!this.enabled || !enableHighSignalNotifications()) {
+      return
+    }
+
+    const subscribedAccounts = this.subscriptions.map(s => s.account)
+
+    // Clear subscriptions for accounts that are no longer in the list
+    for (const account of subscribedAccounts) {
+      if (!accountIncluded(account, accounts)) {
+        this.unsubscribeFromAccount(account)
+      }
+    }
+
+    // Subscribe to new accounts
+    for (const account of accounts) {
+      if (!accountIncluded(account, subscribedAccounts)) {
+        this.subscribeToAccount(account)
+      }
+    }
+  }
+
+  private sessionForAccount(
+    account: Account
+  ): IAliveEndpointSession | undefined {
+    return this.sessionPerEndpoint.get(account.endpoint)
+  }
+
+  private async createSessionForAccount(
+    account: Account
+  ): Promise<IAliveEndpointSession | null> {
+    const session = this.sessionForAccount(account)
+    if (session !== undefined) {
+      return session
+    }
+
+    const api = API.fromAccount(account)
+    const webSocketUrl = await api.getAliveWebSocketURL()
+
+    if (webSocketUrl === null) {
+      return null
+    }
+
+    const aliveSession = new AliveSession(
+      webSocketUrl,
+      api.getAliveWebSocketURL,
+      false,
+      this.notify
+    )
+
+    const newSession = {
+      session: aliveSession,
+      webSocketUrl,
+    }
+
+    this.sessionPerEndpoint.set(account.endpoint, newSession)
+
+    return newSession
+  }
+
+  private unsubscribeFromAccount(account: Account) {
+    const endpointSession = this.sessionForAccount(account)
+    if (endpointSession === undefined) {
+      return
+    }
+
+    const subscription = this.subscriptions.find(s =>
+      accountEquals(s.account, account)
+    )
+    if (subscription === undefined) {
+      return
+    }
+
+    endpointSession.session.unsubscribe([subscription.subscription])
+    this.subscriptions = this.subscriptions.filter(
+      s => !accountEquals(s.account, account)
+    )
+
+    this.sessionPerEndpoint.delete(account.endpoint)
+
+    endpointSession.session.offline()
+
+    console.log('Unubscribed from Alive channel!')
+  }
+
+  private subscribeToAccount = async (account: Account) => {
+    const endpointSession = await this.createSessionForAccount(account)
+    const api = API.fromAccount(account)
+    const channelInfo = await api.getAliveDesktopChannel()
+
+    if (endpointSession === null || channelInfo === null) {
+      return
+    }
+
+    const subscription = {
+      subscriber: this,
+      topic: {
+        name: channelInfo.channel_name,
+        signed: channelInfo.signed_channel,
+        offset: '', // TODO: do we need to set an offset?
```

Comment:
I still have to look into what to do about this, more in upcoming PRs…

Reply:
It seems this argument should be optional and we can ignore it safely 😌

Reply:
S